sdk/rust/src/token/token_supply_type.rs

```diff
@@ -0,0 +1,116 @@
+use std::fmt::{self, Debug, Display, Formatter};
+use std::str::FromStr;
+
+use hedera_proto::services;
+use serde_with::{DeserializeFromStr, SerializeDisplay};
+
+use crate::{Error, FromProtobuf, ToProtobuf};
+
+/// Possible Token Supply Types (IWA Compatibility).
+/// Indicates how many tokens can have during its lifetime.
+#[derive(SerializeDisplay, DeserializeFromStr, Hash, PartialEq, Eq, Clone, Copy)]
```

Comment:
```suggestion
#[derive(Serialize, Deserialize, Hash, PartialEq, Eq, Clone, Copy)]
```

We actually want to lean on normal serde here for enums.